search_engine_parser/core/engines/baidu.py

```diff
@@ -38,7 +38,7 @@ def parse_soup(self, soup):
 		Hence, a regex is used to match all eligible ids
 		"""
 
-		return soup.find_all('div',{'id':re.compile("^\d{1,2}")})
+		return soup.find_all('div',{'id':re.compile(r"^\d{1,2}")})
```

Comment:
Can the offset have only 2 values? 1 and 2

Reply:
After checking out the Baidu search engine, i kinda noticed a pattern in the results. The divs have `id` with respect to the page number. So page `6` divs have ids `7*`. Also all divs have classes `result` or `result-op`. So, we can get this by getting the div with a numeric id using regex and also add a class regex that checks for the presence of `result` or `result-op`. Lemme know what you think

Reply:
Actually, the regex` r"^\d{1,2}" `will match **all** divs with ids that start with a number as \d is shorthand for digits, and {1,2} means 1 or 2 occurrences. In fact, the {1,2} is unnecessary and can be changed to `\d+` (I initially kept it as I was *mistakenly* thinking that a user won't go _beyond_ page 10, but the regex works fine for id=123 as well). Let me know if I should change it to the compact form `r"^\d+" `

In my opinion, the second part is unnecessary. The only **divs** that have id's starting with numbers are the **divs** we are interested in, so it is unnecessary to find the class 
`result/result-op` from it and filter it further, as `c-abstract` class has all the description, and `h3` tags have the titles.